xds/src/test/java/io/grpc/xds/ServerXdsClientNewServerApiTest.java

```diff
@@ -376,6 +376,8 @@ public void notifyUpdatedListener() {
     ArgumentCaptor<ListenerUpdate> listenerUpdateCaptor = ArgumentCaptor.forClass(null);
     verify(listenerWatcher, times(1)).onListenerChanged(listenerUpdateCaptor.capture());
 
+    // expect only 2 requests: original req and ACK: no other req for no other resource
+    verify(requestObserver, times(2)).onNext(any(DiscoveryRequest.class));
```

Comment:
This verify isn't useful. `any(DiscoveryRequest.class)` doesn't include anything, you need to be explicit for things you want to verify (on the other hand, for things not interested in terms of its behavior for a particular test case, you should just not verify it). I don't think we are much interested in verifying the ACK behavior particularly in this test case (that's why it was omitted in the original version). `getSubscribedResources()` is not covered by this test case, it's not called anywhere inside this test case's operation. It should be verified in `streamClosedAndRetry()` test case. One more thing, try to keep the two copies of ServerXdsClientTest consistent (that's why I hate maintaining two copies of tests 😄 ).

Reply:
I have moved the test to `streamClosedAndRetry()` because that's where the new code is exercised. Check the latest commit. Also the count combined with the previous test verifies that LDS was the only query sent on the stream.